dashboard-api/internal/platform/server/handler/segmentsprogress/list_test.go

```diff
@@ -0,0 +1,161 @@
+									Datetime:                    1605049200000,
+									CurrentBudget:               72000,
+									DesiredPercentage:           5,
+									CurrentPercentage:           8.25,
+									ExpectedPercentage:          8.67,
+									DesiredParticipants:         (*int64)(nil),
+									ExpectedParticipants:        64,
+									CurrentParticipants:         59,
+									CurrentPricePerParticipant:  100,
+									PercentageDeviationFromGoal: 3.25,
+								},
+							},
+							Datetime: 1605049200000,
+						},
+					},
+				),
+				nil,
+			)
+
+		res := performRequest("/studies/example-study/segments-progress", studyRepository, studySegmentsRepository)
+
+		assert.Equal(t, http.StatusOK, res.StatusCode)
+		assert.Equal(t, "application/json; charset=utf-8", res.Header.Get("Content-Type"))
+		assert.Equal(t, "{\"data\":[{\"segments\":[{\"id\":\"25-spain-male\",\"name\":\"25-spain-male\",\"datetime\":1605045600000,\"currentBudget\":72000,\"desiredPercentage\":5,\"currentPercentage\":0,\"expectedPercentage\":0,\"desiredParticipants\":null,\"expectedParticipants\":0,\"currentParticipants\":0,\"currentPricePerParticipant\":0,\"percentageDeviationFromGoal\":5}],\"datetime\":1605045600000},{\"segments\":[{\"id\":\"25-spain-male\",\"name\":\"25-spain-male\",\"datetime\":1605049200000,\"currentBudget\":72000,\"desiredPercentage\":5,\"currentPercentage\":8.25,\"expectedPercentage\":8.67,\"desiredParticipants\":null,\"expectedParticipants\":64,\"currentParticipants\":59,\"currentPricePerParticipant\":100,\"percentageDeviationFromGoal\":3.25}],\"datetime\":1605049200000}]}", res.body)
+	})
+
+	t.Run("should return a 200 with an empty array when there are no segments-progress for the specified study", func(t *testing.T) {
+		studyRepository := new(storagemocks.StudyRepository)
+		studySegmentsRepository := new(storagemocks.StudySegmentsRepository)
+		study := studiesmanager.NewStudy("5372ca9c-9fcd-42d4-a596-d90792909917", "Example Study", "example-study", 1605049200000)
+		studyRepository.On("GetStudyBySlug", mock.Anything, "example-study").Return(study, nil)
+		studySegmentsRepository.On("GetAllTimeSegmentsProgress", mock.Anything, study.Id).
+			Return(
+				[]studiesmanager.SegmentsProgress{},
+				nil,
+			)
+
+		res := performRequest("/studies/example-study/segments-progress", studyRepository, studySegmentsRepository)
+
+		assert.Equal(t, http.StatusOK, res.StatusCode)
+		assert.Equal(t, "application/json; charset=utf-8", res.Header.Get("Content-Type"))
+		assert.Equal(t, "{\"data\":[]}", res.body)
+	})
+}
+
+type response struct {
+	StatusCode int
+	Header     http.Header
+	body       string
+}
+
+func performRequest(path string, studyRepository studiesmanager.StudyRepository, studySegmentsRepository studiesmanager.StudySegmentsRepository) response {
+	gin.SetMode(gin.TestMode)
+	r := gin.New()
+	r.GET("/studies/:slug/segments-progress", ListHandler(studyRepository, studySegmentsRepository))
```

Comment:
Why is the path redefined in the tests?

Reply:
Your `Server` struct anyways is given the repositories and has a `RegisterRoutes`. Why don't you use that?

Reply:
Refactored the code to take advantage of it 👌